app/controllers/api/v1/articles_controller.rb

```diff
@@ -0,0 +1,15 @@
+module Api
+  module V1
+    # @note This controller partially authorizes with the ArticlePolicy, in an ideal world, it would
+    #       fully authorize.  However, that refactor would require significantly more work.
+    class ArticlesController < ApiController
+      include Api::ArticlesController
+
+      before_action :validate_article_param_is_hash, only: %i[create update]
+
+      before_action :set_cache_control_headers, only: %i[index show show_by_slug]
+
+      after_action :verify_authorized, only: %i[create]
+    end
```

Comment:
Each `Api::V0/V1::ArticlesController` now only differs in the `before_action`/`after_action` used in them **and** in the way each of their `ApiController` (slightly different in V0/V1 modules). Each action implementation lives in the concern.